src/main/java/ca/uhn/fhir/jpa/starter/common/PartitionModeConfigurer.java

```diff
@@ -0,0 +1,56 @@
+package ca.uhn.fhir.jpa.starter.common;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.jpa.interceptor.PatientIdPartitionInterceptor;
+import ca.uhn.fhir.jpa.model.config.PartitionSettings;
+import ca.uhn.fhir.jpa.partition.PartitionManagementProvider;
+import ca.uhn.fhir.jpa.searchparam.extractor.ISearchParamExtractor;
+import ca.uhn.fhir.jpa.starter.AppProperties;
+import ca.uhn.fhir.rest.server.RestfulServer;
+import ca.uhn.fhir.rest.server.interceptor.partition.RequestTenantPartitionInterceptor;
+import ca.uhn.fhir.rest.server.tenant.UrlBaseTenantIdentificationStrategy;
+import jakarta.annotation.PostConstruct;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class PartitionModeConfigurer {
+	private static final Logger ourLog = LoggerFactory.getLogger(PartitionModeConfigurer.class);
+
+	@Autowired
+	private AppProperties myAppProperties;
+
+	@Autowired
+	private FhirContext myFhirContext;
+
+	@Autowired
+	private ISearchParamExtractor mySearchParamExtractor;
+
+	@Autowired
+	private PartitionSettings myPartitionSettings;
+
+	@Autowired
+	private RestfulServer myRestfulServer;
+
+	@Autowired
+	private PartitionManagementProvider myPartitionManagementProvider;
+
+	@PostConstruct
```

Comment:
Is the use of PostConstruct really needed? It has not been needed for the rest of the starter project

Reply:
@jamesagnew see above regarding PostConstruct.

Reply:
Hmm, I mean of course it's not strictly needed, Spring provides a few ways of injecting dependencies. But this style is the one we use in HAPI FHIR. We have hundreds of classes written this way, and I'd prefer to be consistent.

Reply:
Okay - yes, I know HAPI FHIR is more fond of using that pattern.

The thing is I've tried to eager-bind everything in the starter project from day 1, to make it fail as fast as possible in the starter project - the same goes with initialization. I'm not gonna be picky here. I can always have a look later on to see if there's a way around it.

Reply:
@jkiddo @jamesagnew I sense this is now resolved. Am I OK to resolve this conversation and get an approval?

Reply:
Its ok for me